src/extensions/api/index.ts

```diff
@@ -704,4 +705,93 @@ express.post('/api/pause/:slackId', limiter, async (req, res) => {
     } catch (error) {
         emitter.emit('error', { error });
     }
+});
+/**
+ * Set the goal for a session
+ */
+express.post('/api/setGoal/:slackId', limiter, async (req, res) => {
+    try {
+        if (!req.apiKey) {
+            return res.status(401).send({
+                ok: false,
+                error: 'Unauthorized',
+            });
+        }
+
+        const goalId = req.body.id;
+
+        if (!goalId || typeof goalId !== 'string' || goalId.length === 0) {
+            return res.status(400).send({
+                ok: false,
+                error: 'Missing or invalid id parameter',
+            });
+        }
+
+        const apiKey = scryptSync(req.apiKey, 'salt', 64).toString('hex');
+
+        var session = await prisma.session.findFirst({
+            where: {
+                user: {
+                    apiKey
+                },
+                completed: false,
+                cancelled: false,
+            },
+            include: {
+                user: {
+                    select: {
+                        slackUser: {
+                            select: {
+                                slackId: true
+                            }
+                        }
+                    }
+                }
+            }
+        });
+
+        if (!session || !session.user.slackUser?.slackId) {
+            return res.status(401).send({
+                ok: false,
+                error: 'Invalid user or no active session found',
+            });
+        }
+
+        const newGoal = await prisma.goal.findUniqueOrThrow({
+            where: {
+                id: goalId
+            }
+        });
```

Comment:
I think it would be more helpful to send a 400 response here instead of throwing. This is the first time the goal id is validated, so not being able to find the goal isn't a problem with the api, it's an issue with the request's payload

Reply:
Good catch! Will revise 👍